telegram/ext/_dispatcher.py

```diff
@@ -573,11 +570,43 @@ def add_handler(self, handler: Handler[UT, CCT], group: int = DEFAULT_GROUP) ->
 
         if group not in self.handlers:
             self.handlers[group] = []
-            self.groups.append(group)
-            self.groups = sorted(self.groups)
 
         self.handlers[group].append(handler)
 
+    def add_handlers(
+        self,
+        handlers: Union[Sequence[Handler], Dict[int, Sequence[Handler]]],
+        group: int = DEFAULT_GROUP,
+    ) -> None:
+        """Register multiple handlers at once. Priority/order of handlers matters.
+
+        .. versionadded:: 14.0
+
+        Args:
+            handlers (Sequence[:obj:`telegram.ext.Handler`] | \
+                Dict[int, Sequence[:obj:`telegram.ext.Handler`]]): \
+                Specify a sequence of handlers *or* a dictionary where the keys are groups and
+                values are handlers.
+            group (:obj:`int`, optional): Specify which group the sequence of ``handlers``
+                should be added to. Defaults to ``0``.
+
+        .. seealso:: :meth:`add_handler`
+        """
+        if isinstance(handlers, dict) and group != DEFAULT_GROUP:
+            raise ValueError('The `group` argument can only be used with a sequence of handlers.')
+
+        if isinstance(handlers, list):
```

Comment:
this will fail, if you pass a tuple or another `Sequence` type. A quick solution would be to just swap to

```python
if isinstance(handlers, dict):
    ...
else:
```

Or, if you want to make it as general as possible, you could use `typing.Mapping` instead of `typing.Dict` - then you'd have to use use either `instance(handlers, collections.abc.Sequence)` or `… abc.Mapping)`.
But I'm fully okay with annotation as `Dict` for the mapping - I'd also be okay with annotating just as `Union[Tuple, List]` instead of `Sequence`.